rpc/v1/publish_test.go

```diff
@@ -0,0 +1,206 @@
+			fields: fields{
+				topicName: "walrus",
+			},
+			wantErr: &grpcError{
+				code: codes.InvalidArgument,
+				msg:  vos.ErrEmptyTopicName,
+			},
+		},
+		{
+			name: "short published by message should error",
+			args: args{
+				ctx: context.Background(),
+				message: vos.Message{
+					TopicName:   "walrus",
+					PublishedBy: "fb",
+					Body:        []byte("hello world"),
+				},
+			},
+			fields: fields{
+				topicName: "walrus",
+			},
+			wantErr: &grpcError{
+				code: codes.InvalidArgument,
+				msg:  vos.ErrPublishedByTooShort,
+			},
+		},
+		{
+			name: "empty published by message should error",
+			args: args{
+				ctx: context.Background(),
+				message: vos.Message{
+					TopicName:   "walrus",
+					PublishedBy: "",
+					Body:        []byte("helloorld"),
+				},
+			},
+			fields: fields{
+				topicName: "walrus",
+			},
+			wantErr: &grpcError{
+				code: codes.InvalidArgument,
+				msg:  vos.ErrEmptyPublishedBy,
+			},
+		},
+		{
+			name: "nonexistent topic should return error",
+			args: args{
+				ctx: context.Background(),
+				message: vos.Message{
+					TopicName:   "elephant",
+					PublishedBy: "noone",
+					Body:        []byte("bye world"),
+				},
+			},
+			fields: fields{
+				topicName: "walrus",
+			},
+			wantErr: &grpcError{
+				code: codes.NotFound,
+				msg:  entities.ErrTopicNotFound,
+			},
+		},
+	}
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			// setup server
+			buffer := 1024 * 1024
+			ctx := context.Background()
+			lis := bufconn.Listen(buffer)
+			srv := grpc.NewServer()
+
+			conn, err := grpc.DialContext(ctx, "bufnet", grpc.WithContextDialer(func(context.Context, string) (net.Conn, error) {
+				return lis.Dial()
+			}), grpc.WithInsecure())
+			require.NoError(t, err)
+			defer conn.Close()
+
+			// mock use case
+			mockedUseCase := &usecases.UseCaseMock{
+				PublishFunc: func(ctx context.Context, message vos.Message) error {
+					if err := message.Validate(); err != nil {
+						return err
+					}
+					if tt.fields.topicName != tt.args.message.TopicName {
+						return entities.ErrTopicNotFound
+					}
+					return nil
+				},
+			}
+
+			// create usecase
+			useCaseRPC := New(mockedUseCase, logrus.NewEntry(&logrus.Logger{}))
+
+			// serve listener
+			pb.RegisterWalrusServer(srv, useCaseRPC)
+			go func() {
+				err := srv.Serve(lis)
+				require.NoError(t, err)
+			}()
+
+			// init client
+			client := pb.NewWalrusClient(conn)
+
+			// publish to topic
+			publishRequest := pb.PublishRequest{
+				Message: &pb.Message{
+					Topic:       tt.args.message.TopicName.String(),
+					PublishedBy: tt.args.message.PublishedBy,
+					Body:        tt.args.message.Body,
+				},
+			}
+			_, err = client.Publish(tt.args.ctx, &publishRequest)
+			if *tt.wantErr != (grpcError{}) {
+				s, _ := status.FromError(err)
+				assert.Equal(t, *tt.wantErr, grpcError{code: s.Code(), msg: fmt.Errorf("%s", s.Message())})
+				return
+			}
+			assert.NoError(t, err)
```

Comment:
Your PR are not passing on CI due to complexity, you need to simplify this code
```suggestion
                      assert.ErrorIs(t, err, tt.wantErr)
			if tt.wantErr != nil {
				gotCode, _ := status.FromError(err)
				assert.Equal(t, tt.wantCode, gotCode)
				return
			}
```